src/axom/sidre/nanobind_sidre.cpp

```diff
@@ -161,6 +191,17 @@ NB_MODULE(pysidre, m_sidre)
     .value("DOUBLE_ID", DOUBLE_ID)
     .export_values();
 
+  // Bindings to support iterating collections
+  using AttributeIterator = axom::IndexedCollection<Attribute>::iterator_adaptor;
+  using BufferIterator = axom::IndexedCollection<Buffer>::iterator_adaptor;
+  using GroupIterator = axom::IndexedCollection<Group>::iterator_adaptor;
+  using ViewIterator = axom::IndexedCollection<View>::iterator_adaptor;
+
+  bindIterator<AttributeIterator>(m_sidre, "AttributeIterator");
+  bindIterator<BufferIterator>(m_sidre, "BufferIterator");
+  bindIterator<GroupIterator>(m_sidre, "GroupIterator");
+  bindIterator<ViewIterator>(m_sidre, "ViewIterator");
```

Comment:
Creates bindings for Sidre's iterator types using the `bindIterator` helper function.